include/deal.II/particles/particle_handler.h

```diff
@@ -0,0 +1,480 @@
+       * the function can store additional data. The function is expected to
+       * return a void pointer pointing to a position right after its data
+       * block.
+       * @param load_callback A function that is called once per particle
+       * when deserializing particle data. Arguments to the function are a
+       * particle iterator that identifies the current particle and a void
+       * pointer that points to a data block of size size_callback() in which
+       * additional data was stored by the store_callback function. The
+       * function is expected to return a void pointer pointing to a position
+       * right after its data block.
+       */
+      void
+      register_additional_store_load_functions(const std::function<std::size_t ()> &size_callback,
+                                               const std::function<void *(const particle_iterator &,
+                                                                                void *)> &store_callback,
+                                               const std::function<const void *(const particle_iterator &,
+                                                                                      const void *)> &load_callback);
+
+      /**
+       * Return the total number of particles that were managed by this class
+       * the last time the update_n_global_particles() function was called.
+       * The actual number of particles may have changed since then if
+       * particles have been added or removed.
+       *
+       * @return Total number of particles in simulation.
+       */
+      types::particle_index n_global_particles() const;
+
+      /**
+       * Return the number of particles in the local part of the
+       * triangulation.
+       */
+      types::particle_index n_locally_owned_particles() const;
+
+      /**
+       * Return the number of properties each particle has.
+       */
+      unsigned int n_properties_per_particle() const;
+
+      /**
+       * Return a reference to the property pool that owns all particle
+       * properties, and organizes them physically.
+       */
+      PropertyPool &
+      get_property_pool() const;
+
+      /**
+       * Return the number of particles in the given cell.
+       */
+      unsigned int
+      n_particles_in_cell(const typename Triangulation<dim,spacedim>::active_cell_iterator &cell) const;
+
+      /**
+       * Returns a vector that contains a tensor for every vertex-cell
+       * combination of the output of dealii::GridTools::vertex_to_cell_map()
+       * (which is expected as input parameter for this function).
+       * Each tensor represents a geometric vector from the vertex to the
+       * respective cell center.
+       */
+      std::vector<std::vector<Tensor<1,spacedim> > >
+      vertex_to_cell_centers_directions(const std::vector<std::set<typename parallel::distributed::Triangulation<dim,spacedim>::active_cell_iterator> > &vertex_to_cells) const;
```

Comment:
from the description, it does not look like this belongs to `ParticleHandler`. maybe it should be in `GridTools` or alike.